core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -307,6 +307,10 @@ public void initialise() throws InitialisationException {
 
         lifecycleStrategy.initialise(this);
 
+        // if there is a correlation id generator, try generating one. If there is an error in the generation
+        // we want to catch it in deploy time, and avoid waiting for the first event to appear in runtime
+        getConfiguration().getDefaultCorrelationIdGenerator().ifPresent(generator -> generator.generateCorrelationId());
```

Comment:
use a more explicit script validation instead of this

Reply:
But Validation will return an object with a failure, which won't have the error itself (I mean, exception). So I can't just raise the correct exception (to prevent the app to be deployed)